src/pages/TransactionInfoPage.tsx

```diff
@@ -115,138 +115,131 @@ const TransactionInfoPage = () => {
   return (
     <Section>
       <SectionTitle title="Transaction" />
-      {!loading ? (
-        <>
-          {txInfo && txInfoStatus === 200 && txInfo ? (
-            <Table bodyOnly>
-              <TableBody>
-                <TableRow>
-                  <td>Hash</td>
-                  <HighlightedCell textToCopy={txInfo.hash}>{txInfo.hash}</HighlightedCell>
-                </TableRow>
+      {!txInfoError ? (
+        <Table bodyOnly isLoading={loading} minHeight={450}>
+          {txInfo && (
+            <TableBody>
+              <TableRow>
+                <td>Hash</td>
+                <HighlightedCell textToCopy={txInfo.hash}>{txInfo.hash}</HighlightedCell>
+              </TableRow>
+              <TableRow>
+                <td>Status</td>
+                {isTxConfirmed(txInfo) ? (
+                  <td>
+                    <Badge
+                      type="plus"
+                      content={
+                        <span>
+                          <Check style={{ marginRight: 5 }} size={15} />
+                          Success
+                        </span>
+                      }
+                      inline
+                    />
+                  </td>
+                ) : (
+                  <td>
+                    <Badge
+                      type="neutral"
+                      content={
+                        <>
+                          <LoadingSpinner style={{ marginRight: 5 }} size={15} />
+                          <span>Pending</span>
+                        </>
+                      }
+                    />
+                  </td>
+                )}
+              </TableRow>
+              {isTxConfirmed(txInfo) && txInfo.blockHash && txBlock && (
                 <TableRow>
-                  <td>Status</td>
-                  {isTxConfirmed(txInfo) ? (
-                    <td>
+                  <td>Block</td>
+                  <td>
+                    <SimpleLink
+                      to={`../blocks/${txInfo.blockHash || ''}`}
+                      data-tip={`On chain ${txChain?.chainFrom} → ${txChain?.chainTo}`}
+                    >
+                      {txBlock?.height.toString()}
```

Comment:
If `txBlock` is `undefined` the link will have no text

If you're certain this is not undefined then https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator is the operator to use, not the optional chaining operator, to satisfy the typechecker